### ISSUE TITLE
Question: overlayWith (watermark) - dynamic size

### ISSUE THREAD

**Issue by matthewnessworthy** [lovell/sharp#736]
Is it possible to compose/overlay a watermark image over an image that is being resized and have the watermark take space relative to the final image size?

```
let transformer = sharp();
transformer.background({r: 255, g: 255, b: 255, alpha: 0});
transformer.flatten();
transformer.extend(1);
transformer.trim(1);
transformer.resize(200, 200);
transformer.max();

return transformer;
```

Assuming the source image that is being resized is 200x400, the `.resize(200, 200).max()` would result in a final image with the dimensions of 100x200. 
How should I go about placing an overlay that takes up a max of 30% width and 40% height, centered over the resized image?

Side note: given `.background({r: 255, g: 255, b: 255, alpha: 0}).extend(1).trim(1)`, the resized image could potentially be only 10x10 (if there's a large amount of `#fff` at the source image's edge)

Basically, something similar to http://glide.thephpleague.com/1.0/api/watermarks/

Edit: Currently everything works on a stream, and works perfectly, except for the watermarking capabilities

**Comment by lovell** [comment]
Hello, if you have an input Stream and need to make calculations for `overlayWith` parameters based on the dimensions of the image it provides then #236 is the future enhancement for you.

In the meantime, perhaps look at some of the issues linked to from it for the various workarounds others have suggested.

**Comment by matthewnessworthy** [comment]
Thanks for the update. I looked around and didn't find a solution that acted upon a stream. Any idea when #236 might be worked on?

**Comment by lovell** [comment]
If you expect #236 will meet your needs then please subscribe to it for updates.

**Comment by matthewnessworthy** [comment]
It might work, although handling `.extend(1).trim(1)` might be tricky since there's not way of getting the final image size as part of a stream afaik. I might possibly need to output `.raw().toBuffer()` get the final size, resize an overlay, then using the image buffer as an input argument for a new `sharp()` instance where i can overlay the watermark
Seems fairly CPU & Memory intensive though

Any alternative suggestions are more than welcome.
